### PR TITLE
feat(combat): replace the combat tab with one made for v12

### LINKED ISSUE #526: [V12] Group Initiative Plain Doesn't Work
### What happened?

In short, grouped initiative is _completely_ broken in v12 to the point where you can't begin a combat.

I don't know anyone actively using v12 to run an OSE campaign, so I could be missing something important or there could be workarounds for this and the individual init glitch (issue pending) I don't know of. If not, the combination of both problems is a bad enough show stopper that I think it's inaccurate to claim the system is v12 compatible.

This happens both on a copy of my current campaign world and in a completely fresh OSE world with no mods enabled.

Reproduction steps:

1. Set Initiative to Grouped. This behavior occurs _regardless of the Initiative Persistence setting_. Mine was set to Keep while I did this.
2. Create a new combat and add some PCs and monsters. Set their groups appropriately.
3. Click the Begin Combat button.
4. Initiative is (re)rolled automatically, regardless of whether it was already rolled.
5. An error notification (pictured) appears, and combat does not begin.
group-error-notif
6. Repeat steps 3 through 5 as many times as you like; initiative is rerolled each time, but combat never begins

### What is the expected behaviour?

When Begin Combat is clicked, combat should begin and the first combatant's turn should come up. Also, initiative should probably not be rerolled if it was already rolled before combat began, especially with Initiative Persistence set to Keep.

### Relevant Errors and Warnings

```shell
Cannot assign to read only property '_id' of object '#<Combatant>'

TypeError: Cannot assign to read only property '_id' of object '#<Combatant>'
    at SchemaField._cleanType (foundry-esm.js:7847:20)
    at SchemaField.clean (foundry-esm.js:7342:19)
    at foundry-esm.js:8838:41
    at Array.map (<anonymous>)
    at EmbeddedCollectionField._cleanType (foundry-esm.js:8838:20)
    at EmbeddedCollectionField.clean (foundry-esm.js:7342:19)
    at SchemaField._cleanType (foundry-esm.js:7847:28)
    at SchemaField.clean (foundry-esm.js:7342:19)
    at Combat.cleanData (foundry-esm.js:10295:26)
    at Combat.validate (foundry-esm.js:10407:48)
```

### Additional Support Details

Foundry Virtual Tabletop: Version 12 Stable, 12.328
Game System: ose, 1.10.3
Active Modules: 0
Performance Mode: 1

OS: Windows 10.0.0 x86 (64-bit)
Client: Not/A)Brand/8.0.0.0
GPU: ANGLE (Intel, Intel(R) UHD Graphics 630 (0x00003E9B) Direct3D11 vs_5_0 ps_5_0, D3D11)
Max Texture Size: 16384

Scene: 3840 x 1920 | Grid Size: 100 | Padding Percentage: 0
Walls: 0 | Ambient Lights: 5 | Ambient Sounds: 0 | Tiles: 1 | Tokens: 5
Largest Texture: 3840 x 1920 

Actors: 4 | Items: 0 | Journal Entries: 0 | Rollable Tables: 0 |
Playlists: 0 | Compendium Packs: 0 | Chat Messages: 4

World Scripts: 

### Foundry VTT Core Version

None

### Old-School Essentials Core Version

_No response_

### Please update your browser regularly. Which (updated) browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #527: [V12] Individual Initiative is Rerolled Constantly
### What happened?

Summary: Individual initiative is rerolled, for everyone in the combat, each time anyone advances their turn

I don't know anyone actively using v12 to run an OSE campaign, so I could be missing something important or there could be workarounds for this and the group init glitch I don't know of. If not, the combination of both problems is a bad enough show stopper that I think it's inaccurate to claim the system is v12 compatible.

Reproduction steps:

1. Set initiative to Individual. This glitch happens regardless of the Initiative Persistence setting.
2. Create a new combat, add some PCs and/or monsters, and begin the combat.
3. Advance or rewind the combat turn. Manually setting the active character does not trigger the bug.
4. Initiative is immediately auto-rerolled for **all** combatants, mid-round, and the list is reordered.
5. The only time this doesn't happen seems to be at the end of a round (i.e. when the last combatant in initiative ends their turn). In that case, it appears to respect the Initiative Persistence setting.

### What is the expected behaviour?

Initiative should reset or be rerolled at the beginning of each round, instead of after every single combatant's action.

### Relevant Errors and Warnings

```shell
There is no console output associated with this problem.
```

### Additional Support Details

Foundry Virtual Tabletop: Version 12 Stable, 12.328
Game System: ose, 1.10.3
Active Modules: 0
Performance Mode: 1

OS: Windows 10.0.0 x86 (64-bit)
Client: Not/A)Brand/8.0.0.0
GPU: ANGLE (Intel, Intel(R) UHD Graphics 630 (0x00003E9B) Direct3D11 vs_5_0 ps_5_0, D3D11)
Max Texture Size: 16384

Scene: 3840 x 1920 | Grid Size: 100 | Padding Percentage: 0
Walls: 0 | Ambient Lights: 5 | Ambient Sounds: 0 | Tiles: 1 | Tokens: 5
Largest Texture: 3840 x 1920 

Actors: 4 | Items: 0 | Journal Entries: 0 | Rollable Tables: 0 |
Playlists: 0 | Compendium Packs: 0 | Chat Messages: 34

World Scripts: 

### Foundry VTT Core Version

None

### Old-School Essentials Core Version

_No response_

### Please update your browser regularly. Which (updated) browsers are you seeing the problem on?

Firefox, Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wyrmisis**
## Intent
This changeset completely replaces the existing initiative/combat tracker code with code that uses v12's features for overriding existing pieces of Foundry. It does so by replacing hooks and HTML replacement with new classes for combats, combatants, and the combat tracker sidebar.

It also, if merged, will completely break compatibility with v11; whatever the new release of the system is will need to be for v12+.

###  Group Initiative, Player View
image

### Group Initiative, GM View
image

### Individual Initiative, Player View
image

### Individual Initiative, GM View
image

## Related issues
- fixes #526
- fixes #527

### CODE REVIEW

File: src/templates/sidebar/combat-tracker-combatant-group.hbs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                        <i class="fa-solid fa-arrows-to-eye"></i>
+                    </a>
+                    {{/unless}}
+                    <div class="token-effects">
+                        {{#each this.effects}}
+                        <img class="token-effect" src="{{this}}"/>
+                        {{/each}}
+                    </div>
+                </div>
+            </div>
```
Comment (wyrmisis) [inline src/templates/sidebar/combat-tracker-combatant-group.hbs:65]: This is shared between the two initiative modes and could, admittedly, be moved out into a partial.

**Review by Rughalt — APPROVED** [review]
Fixes the major V12 blocker, will need several UX improvements in separate ticket.

**Comment by wyrmisis** [comment]
> Fixes the major V12 blocker, will need several UX improvements in separate ticket.

If this thing needs UX improvements, that doesn't sound like it should be improved. Would you mind highlighting the more egregious issues, so we can get them fixed before this goes out?

**Comment by Rughalt** [comment]
It's more like janky UX, that we can polish later

**Pull request merged**